packages/lib-classifier/src/components/Classifier/ClassifierContainer.js

```diff
@@ -107,27 +126,35 @@ export default function ClassifierContainer({
     }
   }
 
-  const storedWorkflow = classifierStore.workflows.resources.get(workflowID)
-  if (workflowSnapshot?.id && workflowStrings) {
-    workflowSnapshot.strings = workflowStrings
+  /*
+    If a workflow is stored in session storage... ?
+  */
+ const storedWorkflow = classifierStore.workflows.resources.get(workflowID)
+
+ if (workflowSnapshot?.id && workflowStrings) {
+   workflowSnapshot.strings = workflowStrings
     if (!storedWorkflow) {
       classifierStore.workflows.setResources([workflowSnapshot])
     }
   }
 
+  /*
+    Re-render workflow strings (translations) when workflow id or locale changes
+  */
   useEffect(function onWorkflowStringsChange() {
     if (storedWorkflow && workflowStrings) {
       console.log('Refreshing workflow strings', storedWorkflow.id)
       applySnapshot(storedWorkflow.strings, workflowStrings)
     }
   }, [storedWorkflow, workflowStrings])
```

Comment:
@eatyourgreens What would be a good way to explain these lines of code? I understand when the useEffect is intended to run for changes to the workflow id or if the locale changes.

However, I don't quite know how to explain lines 132-139

Reply:
`classifierStore` is the MobX store, not session storage. Does that help? These lines are updating the workflow in the store, if it’s stale.

Reply:
No wait, putting the workflow in the MobX store if it isn’t already stored.

Reply:
Yeah, those lines should run when you first fetch a workflow from Panoptes and it isn’t already in the store. 

There’s another block, in `Classifier`, which will overwrite the stored workflow if the version changes. You can test that by opening a classifier in one tab, then editing the workflow in another tab. The classifier should refresh with the changes. SWR takes care of fetching new data when the stored object becomes stale. That was my main reason for using it. Our old data-fetching code had no cache validation or refresh for stale data.

Reply:
Just above this is a block that puts the project into the MobX store, when the classifier first loads and the store is empty. 

These lines do the same thing, but store the workflow.

Reply:
We also do this in this particular order, because setting the active project deletes any stored workflows. 

That also prevents us from storing the project and workflow at the same time.

Reply:
Here's an example of this pattern from the React docs, where it's used to derive state from props:
https://react.dev/reference/react/useState#storing-information-from-previous-renders

Worth noting that those docs says this:
>This pattern can be hard to understand and is usually best avoided. However, it’s better than updating state in an effect. When you call the set function during render, React will re-render that component immediately after your component exits with a return statement, and before rendering the children. This way, children don’t need to render twice. The rest of your component function will still execute (and the result will be thrown away). If your condition is below all the Hook calls, you may add an early return; to restart rendering earlier.

I'd assume the same comments apply to changing the store context during a render. Storing the workflow will trigger renders in child components that depend on the workflow.

Reply:
This make sense thanks! I'll add a note about the order of storing a project then workflow, too.